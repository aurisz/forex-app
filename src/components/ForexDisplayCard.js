import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardBody, Row, Col, Button } from 'reactstrap';

import { getExchangeRates, currencyFormatter } from '../utils';

const ForexDisplayCard = ({
  currency,
  baseValue,
  exchangeValue,
  onDeleteCurrency
}) => (
  <Card className="forex-card">
    <CardBody>
      <Row>
        <Col sm={10} className="forex-column-left">
          <Row>
            <Col>
              <h4>{currency}</h4>
            </Col>
            <Col>
              <p className="exchange-value">
                {getExchangeRates(baseValue, exchangeValue)}
              </p>
            </Col>
          </Row>
          <Row>
            <Col>
              <p>
                1 USD = {currency} {currencyFormatter(exchangeValue)}
              </p>
            </Col>
          </Row>
        </Col>

        <Col sm={2} className="forex-column-right">
          <Button
            className="btn-delete-currency"
            color="default"
            block
            onClick={() => onDeleteCurrency(currency)}
            title="Delete Currency"
          >
            <span>(-)</span>
          </Button>
        </Col>
      </Row>
    </CardBody>
  </Card>
);

ForexDisplayCard.propTypes = {
  currency: PropTypes.string.isRequired,
  baseValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  exchangeValue: PropTypes.number.isRequired,
  onDeleteCurrency: PropTypes.func.isRequired
};

export default ForexDisplayCard;
