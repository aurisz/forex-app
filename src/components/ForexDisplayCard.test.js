import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ForexDisplayCard from './ForexDisplayCard';

describe('<ForexDisplayCard />', () => {
  const testProps = {
    currency: 'IDR',
    currencyName: 'Indonesian Rupiah',
    baseValue: 10,
    exchangeValue: 13928,
    onDeleteCurrency: jest.fn()
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<ForexDisplayCard {...testProps} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<ForexDisplayCard {...testProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('delete currency button should work', () => {
    const clickFn = jest.fn();

    const wrapper = shallow(
      <ForexDisplayCard {...testProps} onDeleteCurrency={clickFn} />
    );
    wrapper.find('Button#deleteCurrency').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
