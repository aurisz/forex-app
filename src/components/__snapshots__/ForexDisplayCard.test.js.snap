// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForexDisplayCard /> renders correctly 1`] = `
<div
  className="forex-card card"
>
  <div
    className="row"
  >
    <div
      className="forex-column-left col-9 col-sm-11"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h4>
              IDR
            </h4>
          </div>
          <div
            className="col"
          >
            <h4
              className="exchange-value"
            >
              139,280.00
            </h4>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h6>
              <em>
                IDR
                 - 
                Indonesian Rupiah
              </em>
            </h6>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <span>
              1 USD = 
              IDR
               
              13,928.00
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="forex-column-right col-3 col-sm-1"
    >
      <button
        aria-label={null}
        className="btn-delete-currency btn btn-default btn-block"
        id="deleteCurrency"
        onClick={[Function]}
        title="Delete Currency"
        type="button"
      >
        <span>
          (-)
        </span>
      </button>
    </div>
  </div>
</div>
`;
